devtools/src/test/scala/kalix/devtools/impl/RuntimeVersionCheckerSpec.scala

```diff
@@ -0,0 +1,179 @@
+
+      val lines = readLines(kalixRuntimeFile)
+      val outdatedVersions =
+        RuntimeVersionChecker.collectOutdatedVersions("1.1.28", lines)
+      outdatedVersions shouldBe empty
+    }
+
+    "not collect version from kalix-runtime image if more recent" in {
+      val kalixRuntimeFile =
+        """
+          |version: "3"
+          |services:
+          |  kalix-runtime:
+          |    image: gcr.io/kalix-public/kalix-runtime:1.1.29
+          |    ports:
+          |      - "9000:9000"
+          |""".stripMargin
+
+      val lines = readLines(kalixRuntimeFile)
+      val outdatedVersions =
+        RuntimeVersionChecker.collectOutdatedVersions("1.1.28", lines)
+      outdatedVersions shouldBe empty
+    }
+
+    "collect version from kalix-proxy image if outdated" in {
+      val kalixRuntimeFile =
+        """
+          |version: "3"
+          |services:
+          |  kalix-runtime:
+          |    image: gcr.io/kalix-public/kalix-proxy:1.1.27
+          |    ports:
+          |      - "9000:9000"
+          |""".stripMargin
+
+      val lines = readLines(kalixRuntimeFile)
+      val outdatedVersions =
+        RuntimeVersionChecker.collectOutdatedVersions("1.1.28", lines)
+      outdatedVersions.head shouldBe "gcr.io/kalix-public/kalix-proxy:1.1.27"
+    }
+
+    "not collect version from kalix-proxy image if same" in {
+      val kalixRuntimeFile =
+        """
+          |version: "3"
+          |services:
+          |  kalix-runtime:
+          |    image: gcr.io/kalix-public/kalix-proxy:1.1.28
+          |    ports:
+          |      - "9000:9000"
+          |""".stripMargin
+
+      val lines = readLines(kalixRuntimeFile)
+      val outdatedVersions =
+        RuntimeVersionChecker.collectOutdatedVersions("1.1.28", lines)
+      outdatedVersions shouldBe empty
+    }
+
+    "not collect version from kalix-proxy image if more recent" in {
+      val kalixRuntimeFile =
+        """
+          |version: "3"
+          |services:
+          |  kalix-runtime:
+          |    image: gcr.io/kalix-public/kalix-proxy:1.1.29
+          |    ports:
+          |      - "9000:9000"
+          |""".stripMargin
+
+      val lines = readLines(kalixRuntimeFile)
+      val outdatedVersions =
+        RuntimeVersionChecker.collectOutdatedVersions("1.1.28", lines)
+      outdatedVersions shouldBe empty
+    }
+
+    "collect only version out of date version" in {
+      val kalixRuntimeFile =
+        """
+          |version: "3"
+          |services:
+          |  kalix-runtime:
+          |    image: gcr.io/kalix-public/kalix-runtime:1.1.29
+          |    ports:
+          |      - "9000:9000"
+          |
+          |  kalix-runtime:
+          |    image: gcr.io/kalix-public/kalix-runtime:1.1.27
+          |    ports:
+          |      - "9000:9000"
+          |
+          |  kalix-runtime:
+          |    image: gcr.io/kalix-public/kalix-runtime:1.1.26
+          |    ports:
+          |      - "9000:9000"
```

Comment:
Here I'm taking a shortcut. 

When running multi-services tests, a user may have more than one image in the docker-compose file. We can't know which image will be serving the current application (running locally) and which images will be serving the other services (running in containers). 

So the shortcut is to collect all old versions and recommend to be updated them to the version used to build the current SDK. 

If can be that a service running in a container built with 1.1.28, gets a runtime updated to 1.1.29. The user can choose to ignore the warning or only update it the image for the local run. 

Anyway, this is such a border case that I don't think it's worth to invest more time to make it 100% perfect. Also because the multi-service testing will probably be superseded soonish by `kalix local run`.